python/lsst/eotest/image_utils.py

```diff
@@ -57,52 +57,69 @@ def dm_hdu(hdu):
+    try:
+        imarr = im.Factory(im, overscan).getArray()
+    except AttributeError: # Dealing with a MaskedImage
+        imarr = im.Factory(im, overscan).getImage().getArray()
+    ny, nx = imarr.shape
+    rows = np.arange(ny)
+    values = np.array([statistic(imarr[ii][dxmin:-dxmax]) for ii in rows])
+    return(values)
+
+def bias_func(im, overscan, nskip_cols=5, num_cols=15, **kwargs):
     """Compute the bias by fitting a polynomial (linear, by default)
-    to the mean of each row of the selected overscan region.  This
-    returns a numpy.poly1d object that returns the fitted bias as
+    to the mean of each row of the selected overscan region.  This 
+    returns a numpy.poly1d object that returns the fitted bias as 
     function of pixel row."""
     try:
         imarr = im.Factory(im, overscan).getArray()
     except AttributeError: # Dealing with a MaskedImage
         imarr = im.Factory(im, overscan).getImage().getArray()
     ny, nx = imarr.shape
     rows = np.arange(ny)
-    if fit_order >= 0:
-        values = np.array([statistic(imarr[j]) for j in rows])
-        return np.poly1d(np.polyfit(rows, values, fit_order))
-    else:
-        # Use row-by-row bias level estimate, skipping initial columns
-        # to avoid possible trailed charge.
-        values = np.array([np.median(imarr[j][nskip_cols:nskip_cols+num_cols])
-                           for j in rows])
-        return lambda x: values[int(x)]
+    values = np.array([kwargs.get('statistic')(imarr[j][nskip_cols:nskip_cols+num_cols]) 
+	for j in rows])
+    return np.poly1d(np.polyfit(rows, values, kwargs.get('fit_order'))) 
 
-def bias_image(im, overscan, fit_order=1, statistic=np.mean):
-    my_bias = bias_func(im, overscan, fit_order, statistic=statistic)
+def bias_image(im, overscan, bias_method='func', **kwargs):
+    """Generate a bias image containing the offset values calculated from 
+    bias(), bias_row(), or bias_func()."""
     biasim = afwImage.ImageF(im.getDimensions())
     imarr = biasim.getArray()
     ny, nx = imarr.shape
+    if bias_method not in ['mean', 'row', 'func']:
+        raise RuntimeError('Bias method must be either "mean", "row" or "func".')
+    # Number of rows in overscan must match number of rows in bias image
+    method = {'mean' : bias, 'row' : bias_row, 'func' : bias_func}
+    my_bias = method[bias_method](im, overscan, **kwargs)
+    if bias_method == 'func':
+        my_bias = my_bias(np.arange(ny))
+    elif isinstance(my_bias, float):
+        my_bias = np.full(ny, my_bias)
     for row in range(ny):
-        imarr[row] += my_bias(row)
+        imarr[row] += my_bias[row]
     return biasim
 
 def trim(im, imaging):
     "Trim the prescan and overscan regions."
     return im.Factory(im, imaging)
 
-def unbias_and_trim(im, overscan, imaging,
-                    apply_trim=True, fit_order=1):
+def unbias_and_trim(im, overscan, bias_method, median_stack=None, imaging=None, **kwargs):
```

Comment:
`apply_trim` is redundant here, so replacing that test with `if imaging is not None` is fine, but putting `bias_method` as the third argument breaks backwards compatibility.  So I would suggest instead
```
def unbias_and_trim(im, overscan, imaging=None, bias_method='func', median_stack=None, **kwargs):
```